master/static/migrations/20220311095512_add_workspace_tables.up.sql

```diff
@@ -0,0 +1,25 @@
+CREATE TABLE workspaces (
+  id SERIAL PRIMARY KEY,
+  name TEXT UNIQUE NOT NULL,
+  archived BOOLEAN,
+  user_id INT REFERENCES users(id)
+);
+CREATE TABLE projects (
+  id SERIAL PRIMARY KEY,
+  name TEXT UNIQUE NOT NULL,
+  description TEXT,
+  archived BOOLEAN,
+  workspace_id INT REFERENCES workspaces(id),
+  user_id INT REFERENCES users(id)
+);
+CREATE TABLE notes (
+  id SERIAL PRIMARY KEY,
+  name TEXT NOT NULL,
+  contents TEXT,
+  project_id INT REFERENCES projects(id)
+);
```

Comment:
Definitely not a DB expert but this seems like it's optimizing for pulling projects without notes and then joining when notes are necessary. I looked up some info regarding this and it seems like if project `notes` are `nullable` then querying the table without notes is [just as efficient](https://dba.stackexchange.com/a/18406/228845). cc @stoksc <== DB expert

Reply:
The entire post @hkang1 linked is everything I would say and more. My guess would be a separate table makes sense unless every query that ever queries projects also always wants the notes, and we're sure that will be true basically forever.

Reply:
I mentioned in the PR that I could see this being a JSONB column -- we plan to return the full Project from any API query of Projects. I will switch to JSONB

Reply:
I would recommend keeping them separate, personally.

> we're sure that will be true basically forever.

like for this, i dont quite believe this will be true. e.g., if we do historical allocation aggregation queries and we want to join to projects/workspaces to aggregate over them, then separate would be better since that query won't need notes and postgres is a row store. for context that particular query is already the hardest on the system by a lot.